task_processing/plugins/mesos/execution_framework.py

```diff
@@ -215,15 +215,18 @@ def _reconcile_tasks(self, tasks_to_reconcile):
         self._reconcile_tasks_at += self._task_reconciliation_delay
 
     def offer_matches_pool(self, offer):
```

Comment:
I think we've got an opportunity to shuffle some bits round here.

Can you move this function to ``constraints.py`` and make a function ``pool_match`` just take a pool and offer (it's content can just be ``return equal_op(pool, offer_pool)``).

That way we don't have different logic for the executor pool and the offer pool (you use ``offer_matches_pool`` for the executor pool and an adhoc implementation on line 359.

Then we can have this function in the executor:
```
def offer_matches_pool(self, offer):
  if task.pool:
        return constraints.pool_match(attribute_pool, task.pool)
  elif self.pool:
         return constraints.pool_match(attribute_pool, self.pool)
  else:
        True
```